### PR TITLE
fix: provide raw HTTP response in errors that supports body related m…

### PR DESCRIPTION
**Pull request by TheUnderScorer**
…ethods, such as `response.json()`

### CODE REVIEW

**Review by necipallef — DISMISSED** [review]

File: src/responseUtils.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export function getJson<T>(response: Response) {
```
Comment (JuroUhlar) [inline src/responseUtils.ts:1]: This generic seems useless to me

File: tests/unit-tests/serverApiClientTests.spec.ts [hunk 2]
```diff
@@ -14,4 +14,39 @@ describe('ServerApiClient', () => {
 
     expect(mockFetch).toHaveBeenCalledTimes(1)
   })
+
+  it('errors should return response that supports body related methods', async () => {
+    const mockFetch = jest.fn().mockResolvedValue(
+      new Response(
+        JSON.stringify({
+          status: 403,
```
Comment (JuroUhlar) [inline tests/unit-tests/serverApiClientTests.spec.ts:22]: This is slightly confusing, the API would not return `status` inside response body, right? It returns this

```json
{
  "error": {
    "code": "FeatureNotEnabled",
    "message": "feature not enabled"
  }
}

```

File: tests/unit-tests/serverApiClientTests.spec.ts [hunk 3]
```diff
@@ -14,4 +14,39 @@ describe('ServerApiClient', () => {
...
+          error: {
+            code: 'FeatureNotEnabled',
+            message: 'feature not enabled',
+          },
+        } as Omit<EventError, 'response'>),
```
Comment (JuroUhlar) [inline tests/unit-tests/serverApiClientTests.spec.ts:27]: What is the purpose of this `as` ? 
If you wanted to enforce the expected API response shape, I think you need 

`satisfies Omit<EventError, 'response' | 'status'>`

File: tests/unit-tests/serverApiClientTests.spec.ts [hunk 4]
```diff
@@ -14,4 +14,39 @@ describe('ServerApiClient', () => {
...
+        { status: 403 }
+      )
+    )
+
+    const client = new FingerprintJsServerApiClient({
+      fetch: mockFetch as any,
```
Comment (JuroUhlar) [inline tests/unit-tests/serverApiClientTests.spec.ts:33]: what is the purpose of `as any` here?

**Review by JuroUhlar — APPROVED** [review]
Thx!

**Review by JuroUhlar — APPROVED** [review]

**Review by necipallef — APPROVED** [review]

**Review by makma — APPROVED** [review]

**Pull request merged**